WordPress/Classes/Services/PostRepository.swift

```diff
@@ -256,14 +256,44 @@ final class PostRepository {
         ContextManager.shared.saveContextAndWait(context)
     }
 
+    /// Trashes the given post.
+    ///
+    /// - warning: This method delets all local revision of the post.
+    @MainActor
+    func _trash(_ post: AbstractPost) async throws {
+        assert(post.isOriginal())
+
+        let context = coreDataStack.mainContext
+
+        guard let postID = post.postID, postID.intValue > 0 else {
+            context.deleteObject(post) // Delete all the local data
+            ContextManager.shared.saveContextAndWait(context)
+            return
+        }
+
+        post.deleteAllRevisions()
+        ContextManager.shared.saveContextAndWait(context)
+
+        let remote = try getRemoteService(for: post.blog)
+        var remotePost = try await remote.post(withID: postID)
```

Comment:
I tested it and it only takes a fraction of the time of the actual `/delete` call, so I think it's OK to invoke it every time.